File: 백준/Silver/3986. 좋은 단어/좋은 단어.js
'use strict';
const path = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = require('fs').readFileSync(path).toString().trim().split('\n');

const [N, ...words] = input;

let result = 0;

for (let i = 0; i < N; i++) {
  const word = words[i];
  if (word.length % 2 === 0) {
    const stack = [];
    for (let char of word) {
      stack[stack.length - 1] === char ? stack.pop() : stack.push(char);
    }
    stack.length === 0 && result++;
  }
}

console.log(result);

File: 백준/Silver/3986. 좋은 단어/좋은 단어.py
import sys
input = sys.stdin.readline


N = int(input())
result = 0

for _ in range(N):
  word = input().rstrip()
  stack = []
  for string in word:
    if stack and stack[-1] == string:
      stack.pop()
      is_good_word = True
    else:
      stack.append(string)
  if not stack:
    result += 1
print(result)